app/uk/gov/hmrc/transitmovementsrouter/services/RoutingService.scala

```diff
@@ -0,0 +1,124 @@
+import akka.stream.scaladsl.Flow
+import akka.stream.scaladsl.GraphDSL
+import akka.stream.scaladsl.Keep
+import akka.stream.scaladsl.Sink
+import akka.stream.scaladsl.Source
+import akka.util.ByteString
+import com.google.inject._
+import play.api.Logging
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.transitmovementsrouter.connectors.MessageConnectorProvider
+import uk.gov.hmrc.transitmovementsrouter.models._
+import uk.gov.hmrc.transitmovementsrouter.services.error.RoutingError
+import cats.data.EitherT
+
+import scala.concurrent.Future
+
+@ImplementedBy(classOf[RoutingServiceImpl])
+trait RoutingService {
+
+  def submitDeclaration(
+    movementType: MovementType,
+    movementId: MovementId,
+    messageId: MessageId,
+    payload: Source[ByteString, _]
+  )(implicit hc: HeaderCarrier): EitherT[Future, RoutingError, Unit]
+}
+
+class RoutingServiceImpl @Inject() (messageConnectorProvider: MessageConnectorProvider)(implicit materializer: Materializer)
+    extends RoutingService
+    with XmlParsingServiceHelpers
+    with Logging {
+
+  val office = Sink.head[ParseResult[OfficeOfDeparture]]
+
+  private val officeOfDepartureSink: Sink[ByteString, Future[ParseResult[OfficeOfDeparture]]] = Sink.fromGraph(
+    GraphDSL.createGraph(office) {
+      implicit builder => officeShape =>
+        import GraphDSL.Implicits._
+
+        val xmlParsing: FlowShape[ByteString, ParseEvent]                            = builder.add(XmlParsing.parser)
+        val officeOfDeparture: FlowShape[ParseEvent, ParseResult[OfficeOfDeparture]] = builder.add(XmlParser.officeOfDepartureExtractor)
+        xmlParsing ~> officeOfDeparture ~> officeShape
+
+        SinkShape(xmlParsing.in)
+    }
+  )
+
+  def buildMessage(messageSender: MessageSender): Flow[ByteString, ByteString, NotUsed] =
+    Flow.fromGraph(
+      GraphDSL.create() {
+        implicit builder =>
+          import GraphDSL.Implicits._
+
+          val xmlParsing: FlowShape[ByteString, ParseEvent]         = builder.add(XmlParsing.parser)
+          val insertSenderWriter: FlowShape[ParseEvent, ParseEvent] = builder.add(XmlParser.messageSenderWriter(messageSender))
+          val toByteString                                          = builder.add(XmlWriting.writer)
+          xmlParsing ~> insertSenderWriter ~> toByteString
+
+          FlowShape(xmlParsing.in, toByteString.out)
+      }
+    )
+
+  override def submitDeclaration(
+    movementType: MovementType,
+    movementId: MovementId,
+    messageId: MessageId,
+    payload: Source[ByteString, _]
+  )(implicit hc: HeaderCarrier): EitherT[Future, RoutingError, Unit] = {
+
+    val officeOfDeparture       = payload.toMat(officeOfDepartureSink)(Keep.right).run()
+    implicit val messageSender  = MessageSender(movementId, messageId)
+    implicit val updatedPayload = payload.via(buildMessage(messageSender))
+
+    EitherT(post(officeOfDeparture))
+  }
+
+  //TODO:  Not quite sure what we return here
```

Comment:
We should be returning a 202 Accepted (for now) which is what we are expecting from EIS[-stub]

Reply:
Yep.

Reply:
Forgot to remove the comment!

Reply:
Yes - though in this layer we're just sending back unit at the moment because we have nothing to report other than a success. The controller will return a 202 upon seeing a unit.

In the future, I suspect this will be an `Option[Source[ByteString, _]]` if we have synch responses and can harness the power of streaming responses.